### DISCUSSION TITLE
Is it possible to edit a `Relationship` column definition?

### DISCUSSION THREAD

**Discussion by cesarverdesslots** [Budibase/budibase#6527] [Help] [answered]
How can I edit and change a `Relationship` column definition without delete the column and create it again?

For example I had defined the `Zona_Tecnica` as a `Relationship` column of the `Salas` table an when I press the pen icon to edit the column, this windows is shown:

![image](https://user-images.githubusercontent.com/83496099/176937208-c6ea6acb-f4f2-420f-b1e4-d307759d4245.png)

As you can watch, all the fields are read only, an error it's shown and the `Save Column` button is disabled.

The error `Column name already in use in table Zonas_Tecnicas` it's not an error... is part of the relationship definition.

The same happened when I tried to edit the other side of the relationship... the column `Salas` of the table `Zonas_Tecnicas`:

![image](https://user-images.githubusercontent.com/83496099/176938928-d02e4467-c7dd-4d1c-8a36-18d7cc35c899.png)

What am I doing wrong? Is it a issue of Budibase?

Thanks in advance...

**Comment by melohagan** [comment, marked as answer]
Hi @cesarverdesslots 

Whenever you create a relationship column in one table, it will also create a column in the related table with the name provided.
This name cannot clash with an existing column name.

You can't rename relationship columns unfortunately - you do have to delete and recreate.

**Reply by cesarverdesslots** [reply]
Ok, Thank you @melohaganI, I understand it. Perhaps it could be removed the edit pencil at the right of the relationship column name. Just a suggestion. Thanks for your answer.

:+1:

**Reply by antoine-gu** [reply]
I would appreciate that functionality to change existing columns into relationsips columns. As when Import data from csv i have to create all the columns in advance  right now.
